tests/utils_tests/test_autoreload.py

```diff
@@ -88,12 +89,12 @@ def test_only_new_files(self):
         self.assertFalse(any(f.endswith('.pyc') for f in gen_filenames()))
 
     def test_deleted_removed(self):
-        fd, filepath = tempfile.mkstemp(dir=os.path.dirname(upath(__file__)), suffix='.py')
-        try:
-            _, filename = os.path.split(filepath)
-            import_module('.%s' % filename.replace('.py', ''), package='utils_tests')
-            self.assertIn(npath(filepath), gen_filenames())
-        finally:
-            os.close(fd)
-            os.remove(filepath)
-        self.assertNotIn(filepath, gen_filenames())
+        dirname = tempfile.mkdtemp()
```

Comment:
Indeed, I didn't bother removing the temporary directory (always) and the temporary file (only if the first assertion fails) because that will happen automatically at the end of the test run.